ledger/complete/wal/syncrename.go

```diff
@@ -60,6 +60,11 @@ func (s *SyncOnCloseRenameFile) Close() error {
 		return fmt.Errorf("error while renaming from %s to %s: %w", s.file.Name(), s.targetName, err)
 	}
 
+	err = requestDropFromOSFileCache(s.targetName)
```

Comment:
Why doing it here? Is it when the checkpoint is generated? In order to drop the checkpoint from memory right after the checkpoint is generated?

Reply:
> Is it when the checkpoint is generated? In order to drop the checkpoint from memory right after the checkpoint is generated?

Yes.  This is the `Close()` of the custom checkpoint file writer.  The caller doesn't have access to checkpoint filename, so this is a convenient and safe spot.